homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -1103,6 +1163,84 @@ async def async_set_dry_off(self):
         )
 
 
+class XiaomiAirHumidifierMiot(XiaomiAirHumidifier):
+    """Representation of a Xiaomi Air Humidifier (MiOT protocol)."""
+
+    @property
+    def speed(self):
+        """Return the current speed."""
+        if self._state:
+            if (
+                AirhumidifierMiotOperationMode(self._state_attrs[ATTR_MODE])
+                == AirhumidifierMiotOperationMode.Low
+            ):
+                return SPEED_LOW
+            if (
+                AirhumidifierMiotOperationMode(self._state_attrs[ATTR_MODE])
+                == AirhumidifierMiotOperationMode.Mid
+            ):
+                return SPEED_MEDIUM
+            if (
+                AirhumidifierMiotOperationMode(self._state_attrs[ATTR_MODE])
+                == AirhumidifierMiotOperationMode.High
+            ):
+                return SPEED_HIGH
+
+        return None
+
+    @property
+    def button_pressed(self):
+        """Return the last button pressed."""
+        if self._state:
+            return AirhumidifierPressedButton(
+                self._state_attrs[ATTR_BUTTON_PRESSED]
+            ).name
+
+        return None
+
+    async def async_set_speed(self, speed: str) -> None:
+        """Set the speed of the fan."""
+
+        translated_speed = 0
+
+        if speed == SPEED_LOW:
```

Comment:
See my comment above about mapping. This map could be done as a class variable, and you could reverse it for translation:
```
reverse_mode_mapping = {v: k for k, v in MODE_MAPPING.items()}
return reverse_mode_mapping[speed]
```